### PR TITLE
DOC: Fix the entries of C functions

### PR DESCRIPTION
**Pull request by takanori-pskq**
Related to #16217, #13114. Separated from #16370.

sphinx 2 doesn't support function-style macro, so basically used `function` instead. When the type of parameter is unknown, I dealt with them in the same way as `PyArray_MAX, PyArray_MIN, ...` (`doc/source/reference/c-api/array.rst`, l.3476).

### CODE REVIEW

File: doc/source/reference/c-api/array.rst [hunk 1]
```diff
@@ -819,38 +820,42 @@ General check of Python Type
...
-.. c:function:: PyArray_IsZeroDim(op)
+.. c:function:: int PyArray_IsZeroDim(PyObject *op)
 
     Evaluates true if *op* is an instance of (a subclass of)
     :c:data:`PyArray_Type` and has 0 dimensions.
 
-.. c:function:: PyArray_IsScalar(op, cls)
+.. c:var:: op
+
+.. c:var:: cls
```
Comment (eric-wieser) [inline doc/source/reference/c-api/array.rst:830]: Why are these needed?

File: doc/source/reference/c-api/array.rst [hunk 2]
```diff
@@ -1529,7 +1534,7 @@ Flag checking
 For all of these macros *arr* must be an instance of a (subclass of)
 :c:data:`PyArray_Type`.
 
-.. c:function:: PyArray_CHKFLAGS(PyObject *arr, flags)
+.. c:function:: int PyArray_CHKFLAGS(PyObject *arr, flags)
```
Comment (eric-wieser) [inline doc/source/reference/c-api/array.rst:1537]: `flags` should be `int flag`

File: doc/source/reference/c-api/array.rst (hunk 1 above)
Comment (takanori-pskq) [reply doc/source/reference/c-api/array.rst:830]: sphinx 2 doesn't support the function style macro, so emits reference warnings without these vars. we don't need these in sphinx 3.

Comment (eric-wieser) [reply doc/source/reference/c-api/array.rst:830]: Since this change leads to confusing rendering in the docs, I'd be inclined to save this bit for the Sphinx 3 PR.

File: doc/source/reference/c-api/array.rst (hunk 2 above)
Comment (takanori-pskq) [reply doc/source/reference/c-api/array.rst]: Fixed.

**Review by eric-wieser — APPROVED** [review]
Looks good, just one more stray `var` to clean up so we don't get weird docs like

https://numpy.org/devdocs/reference/c-api/array.html#c.a
